cmd/goci/main.go

```diff
@@ -34,14 +34,25 @@ func (e *ValidationError) Error() string {
 	return e.Message
 }
 
+type GoModFileNotFoundError struct {
+	Message string
+}
+
+func (e *GoModFileNotFoundError) Error() string {
+	return e.Message
+}
+
 func main() {
 	if len(os.Args) < 2 {
 		fmt.Println("requires 1 argument.", usage)
 		os.Exit(1)
 	}
 	mode := os.Args[1]
 	if err := run(mode); err != nil {
-		if _, ok := err.(*ValidationError); ok {
+		if _, ok := err.(*GoModFileNotFoundError); ok {
+			fmt.Println("Go Mod file not found error:", err)
+			os.Exit(3)
```

Comment:
Not sure about the exit code 3. cc @andruwm to double check the intention

Reply:
I think in this case we can just return no error from the validation function instead of returning a cased error and different error code.